redash/serializers/query_result.py

```diff
@@ -19,42 +20,39 @@ def _convert_bool(value):
 
     return value
 
-def _convert_date(value):
-    if not value:
-        return value
-
-    parsed = parse_date(value)
-
-    return parsed.strftime(_convert_format(current_org.get_setting('date_format')))
 
-
-def _convert_datetime(value):
+def _convert_datetime(value, fmt):

```

Comment:
Can't this all be summed up to

```suggestion
def _convert_datetime(value, fmt):
    try:
        parsed = parse_date(value)
        return parsed.strftime(fmt)
    except Exception:
        return value       
```